test/integration/expression/expression.test.ts

```diff
@@ -7,180 +7,220 @@ import {
     const expression = isFunction(fixture.expression) ?
         createPropertyExpression(convertFunction(fixture.expression, spec), spec) :
         createPropertyExpression(fixture.expression, spec);
 
-    const result: { compiled: any; outputs?: any } = {
-        compiled: getCompilationResult(expression)
-    };
-
-    if (result.compiled.result !== 'error') {
-        result.outputs = evaluateExpression(fixture, expression);
+    if (expression.result === 'error') {
+        return {
+            compiled: getCompilationErrorResult(expression.value),
+        };
     }
-
-    return result;
+    return {
+        compiled: getCompilationSuccessResult(expression.value),
+        outputs: fixture.inputs === undefined ? [] : evaluateExpression(fixture.inputs, expression.value),
+    };
 }
 
-function getCompilationResult(expression: Result<StylePropertyExpression, ExpressionParsingError[]>) {
-    const compilationResult = {} as any;
-    if (expression.result === 'error') {
-        compilationResult.result = 'error';
-        compilationResult.errors = expression.value.map((err) => ({
+function getCompilationErrorResult(parsingErrors: ExpressionParsingError[]): CompilationErrorResult {
+    return {
+        result: 'error',
+        errors: parsingErrors.map((err) => ({
             key: err.key,
-            error: err.message
-        }));
-        return compilationResult;
-    }
-
-    const expressionValue = expression.value;
-    const type = (expressionValue as any)._styleExpression.expression.type; // :scream:
-
-    compilationResult.result = 'success';
-    compilationResult.isFeatureConstant = expressionValue.kind === 'constant' || expressionValue.kind === 'camera';
-    compilationResult.isZoomConstant = expressionValue.kind === 'constant' || expressionValue.kind === 'source';
-    compilationResult.type = toString(type);
-
-    return compilationResult;
+            error: err.message,
+        })),
+    };
 }
 
-function evaluateExpression(fixture: ExpressionFixture, expression: Result<StylePropertyExpression, ExpressionParsingError[]>) {
-
-    let availableImages: any[];
-    let canonical: ICanonicalTileID | null;
+function getCompilationSuccessResult(expression: StylePropertyExpression): CompilationSuccessResult {
+    const kind = expression.kind;
+    const type = getStylePropertyExpressionType(expression);
+    return {
+        result: 'success',
+        isFeatureConstant: kind === 'constant' || kind ==='camera',
+        isZoomConstant: kind === 'constant' || kind === 'source',
+        type: toString(type),
+    };
+}
 
-    const evaluationResult: any[] = [];
+function evaluateExpression(inputs: FixtureInput[], expression: StylePropertyExpression) {
+    const type = getStylePropertyExpressionType(expression);
 
-    const expressionValue = expression.value;
-    const type = (expressionValue as any)._styleExpression.expression.type; // :scream:
+    return inputs.map((input) => {
```

Comment:
I think it's better to have a for loop here instead as this is what actually runs the tests, and a `map` here is confusing (to me)...